contracts/contracts/Mincomind.sol

```diff
@@ -94,15 +101,18 @@ contract MincoMind is Reencrypt {
         emit GuessAdded(msg.sender, latestGames[msg.sender], guess);
     }
 
-    function claim() public {
-        Game storage game = games[msg.sender][latestGames[msg.sender]];
-        require(!game.isComplete, "Game is already complete");
-        Clue memory clue = checkGuessResult(msg.sender, latestGames[msg.sender], game.numGuesses - 1);
-
+    function endGame(address memory user) public {        
+        Game storage game = games[user][latestGames[user]];
+        require(!game.isComplete, "Game is already ended");
+        Clue memory clue = checkGuessResult(user, latestGames[user], game.numGuesses - 1);
+        
+        // if the user has not guessed the secret in 10 minutes, the game can be ended by anyone
+        require(block.timestamp - game.timeStarted > MAX_SECONDS_PER_GAME || clue.bulls == 4, "Game is not yet ended");
+        uint8 memory points = clue.bulls == 4 ? 9 - game.numGuesses : 0;
         require(clue.bulls == 4, "Guess is not correct");
 
         game.isComplete = true;
 
-        emit GameWon(msg.sender, latestGames[msg.sender]);
+        emit GameOutcome(msg.sender, latestGames[msg.sender], points);
```

Comment:
Modified this so that anyone can end a game for a user if they're out of time or they've guessed it correctly, this is in prep so that the deposit can be transferred to the pot